src/main/java/com/williamfiset/algorithms/datastructures/queue/IntQueue.java

```diff
@@ -5,50 +5,59 @@
  * is you need to know an upper bound on the number of elements that will be inside the queue at any
  * given time for this queue to work.
  *
- * @author William Fiset, william.alexandre.fiset@gmail.com
+ * @author William Fiset, william.alexandre.fiset@gmail.com, liujingkun, liujkon@gmail.com
  */
 package com.williamfiset.algorithms.datastructures.queue;
 
 public class IntQueue {
 
-  private int[] ar;
-  private int front, end, sz;
+  private int[] data;
+  private int front, end;
+  private int size;
 
   // maxSize is the maximum number of items
   // that can be in the queue at any given time
   public IntQueue(int maxSize) {
-    front = end = 0;
-    sz = maxSize + 1;
-    ar = new int[sz];
+    front = end = size = 0;
+    data = new int[maxSize];
   }
 
   // Return true/false on whether the queue is empty
   public boolean isEmpty() {
-    return front == end;
+    return size == 0;
   }
 
   // Return the number of elements inside the queue
   public int size() {
-    if (front > end) return (end + sz - front);
-    return end - front;
+    return size;
   }
 
   public int peek() {
-    return ar[front];
+    if (size == 0) {
+      throw new RuntimeException("Queue is empty");
+    }
+    front = front % data.length;
```

Comment:
Using the '%' operation is slow, can you refactor to avoid this, preferably without adding any additional if statement branching?